apps/web/src/utils/useClearURLQueryParams.ts

```diff
@@ -18,13 +18,18 @@ export function useClearURLQueryParams(param: string | string[]) {
     if (hasRenderedOnce.current) return;
     const params = new URLSearchParams(urlQuery as Record<string, string>);
     const paramsToClear = typeof param === 'string' ? [param] : param;
+    let paramDeleted = false;
     for (const p of paramsToClear) {
       if (params.has(p)) {
         params.delete(p);
+        paramDeleted = true;
       }
     }
-    // @ts-expect-error we're simply replacing the current page with the same path
-    replace({ pathname, query: params.toString() }, undefined, { shallow: true });
+    if (paramDeleted) {
+      // @ts-expect-error we're simply replacing the current page with the same path
+      replace({ pathname, query: params.toString() }, undefined, { shallow: true });
+    }
+
```

Comment:
Add a new logic that we only replace the router if we deleted something from the `params`. 

Lmk what you think about this @Robinnnnn

Reply:
wow, i'm super surprised this caused an issue 🤔 i would assume shallow-rendering wouldn't cause the page to rerender like that

great catch!